app/components/toDoListController.js

```diff
@@ -0,0 +1,30 @@
+toDoTaskList.controller('ToDoListController', [function() {
+
+  var self = this;
+  self.toDoList = [];
+
+  self.addToDo = function() {
+    if(!self.newToDo || self.newToDo === '') {
+      return;
+    }
+    self.toDoList.push( {'task': self.newToDo, 'completed': false} );
+    self.newToDo = '';
+  };
+
+  self.taskCounter = function () {
+    return self.toDoList.length;
+  };
+
+  self.deleteTask = function(item) {
+    var index = self.toDoList.indexOf(item);
+    self.toDoList.splice(index, 1);
+  };
+
+// convert to a function using filter without underscore js library
+  self.removeCompleteTasks = function() {
+    self.toDoList = _.filter(self.toDoList, function(todo){
```

Comment:
Hmm did you pull in the whole underscore library to do one filter function? Your better than that lollyphant! https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/filter